src/data/nat/digits.lean

```diff
@@ -414,3 +418,30 @@ begin
   rw of_digits_neg_one at t,
   exact t,
 end
+
+lemma digits_len_le_digits_len_succ (b n : ℕ) : (digits b n).length ≤ (digits b (n + 1)).length :=
+begin
+  cases b,
+  { -- base 0
+    cases n; simp },
+  { cases b,
+    { -- base 1
+      simp },
+    { -- base >= 2
+      apply nat.strong_induction_on n,
+      clear n,
+      intros n IH,
+      cases n,
+      { simp },
+      { repeat { rw digits_add_two_add_one },
+        by_cases hdvd : (b.succ.succ) ∣ (n.succ+1),
+        { rw nat.succ_div_of_dvd hdvd,
+          repeat { rw list.length_cons },
+          rw nat.succ_le_succ_iff,
+          apply IH,
+          exact nat.div_lt_self (by linarith) (by linarith) },
+        { rw nat.succ_div_of_not_dvd hdvd,
+          refl } } } }
+end
+
+lemma le_digits_len_le (b n m : ℕ) (h : n ≤ m) : (digits b n).length ≤ (digits b m).length := monotone_of_monotone_nat (digits_len_le_digits_len_succ b) h
```

Comment:
Line is too long! :-)

(Says the pot, calling the kettle black.)

Reply:
Oh can we just have begin exact end instead then?

Reply:
You don't need to go into tactic mode, just put the proof on a new line

```lean
lemma le_digits_len_le (b n m : ℕ) (h : n ≤ m) : (digits b n).length ≤ (digits b m).length := 
monotone_of_monotone_nat (digits_len_le_digits_len_succ b) h
```

Reply:
oooohhhhh that makes a lot more sense